src/supervisor/kona-supervisor/launcher.star

```diff
@@ -0,0 +1,101 @@
+_file = import_module("/src/util/file.star")
+_net = import_module("/src/util/net.star")
+
+_ethereum_package_constants = import_module(
+    "github.com/ethpandaops/ethereum-package/src/package_io/constants.star"
+)
+
+_observability = import_module("/src/observability/observability.star")
+_prometheus = import_module("/src/observability/prometheus/prometheus_launcher.star")
+
+
+DATA_DIR = "/etc/kona-supervisor"
+DEPENDENCY_SET_FILE_NAME = "dependency_set.json"
+
+
+def launch(
+    plan,
+    params,
+    l1_config_env_vars,
+    l2s,
+    jwt_file,
+    deployment_output,
+    observability_helper,
+):
+    supervisor_l2s = [
+        l2 for l2 in l2s if l2.network_id in params.superchain.participants
+    ]
+
+    config = _get_config(
+        plan=plan,
+        params=params,
+        l1_config_env_vars=l1_config_env_vars,
+        l2s=supervisor_l2s,
+        jwt_file=jwt_file,
+        deployment_output=deployment_output,
+        observability_helper=observability_helper,
+    )
+
+    service = plan.add_service(params.service_name, config)
+
+    _observability.register_kona_service_metrics_job(
+        observability_helper,
+        service,
+    )
+
+    return struct(service=service, l2s=supervisor_l2s)
+
+
+def _get_config(
+    plan,
+    params,
+    l1_config_env_vars,
+    l2s,
+    jwt_file,
+    deployment_output,
+    observability_helper,
+):
+    ports = _net.ports_to_port_specs(params.ports)
+    datadir = params.superchain.dependency_set.name
+
+    cmd = ["kona-supervisor"] + params.extra_params
+
+    # apply customizations
+
+    if observability_helper.enabled:
+        _observability.configure_kona_service_metrics(cmd, ports)
+
+    return ServiceConfig(
+        image=params.image,
+        ports=ports,
+        files={
+            DATA_DIR: params.superchain.dependency_set.name,
+            _ethereum_package_constants.GENESIS_DATA_MOUNTPOINT_ON_CLIENTS: deployment_output,
+            _ethereum_package_constants.JWT_MOUNTPOINT_ON_CLIENTS: jwt_file,
+        },
+        env_vars={
+            "DATADIR": "/db",
+            "DEPENDENCY_SET": "{0}/{1}".format(
+                DATA_DIR, params.superchain.dependency_set.path
+            ),
+            "ROLLUP_CONFIG_PATHS": _ethereum_package_constants.GENESIS_DATA_MOUNTPOINT_ON_CLIENTS
```

Comment:
seems like we need to add this arg to kona-supervisor @dhyaniarun1993

Reply:
code is copied from the op-supervisor launcher https://github.com/ethpandaops/optimism-package/blob/main/src/supervisor/op-supervisor/launcher.star